skyportal/models/instrument.py

```diff
@@ -214,6 +372,16 @@ def api_class_obsplan(self):
     def listener_class(self):
         return getattr(facility_apis, self.listener_classname)
 
+    @property
+    def number_of_fields(self):
+        if not self.has_fields:
+            return 0
+        stmt = sa.select(InstrumentField).where(
+            InstrumentField.instrument_id == self.id
+        )
+        count_stmt = sa.select(func.count()).select_from(stmt.distinct())
+        return DBSession().execute(count_stmt).scalar()
+
```

Comment:
Do you need a query here? Given that the instrument has a relationship on fields, I'm wondering if you can't just check the `len(self.fields)` or something similar instead. Worth trying.

Reply:
self.fields will load all that data, which I think we want to avoid.

Reply:
Ok that makes sense. LGTM